hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/S3GatewayHttpServer.java

```diff
@@ -17,30 +17,90 @@
  */
 package org.apache.hadoop.ozone.s3;
 
+import com.google.common.base.Strings;
 import java.io.IOException;
-
-import org.apache.hadoop.hdds.conf.MutableConfigurationSource;
-import org.apache.hadoop.hdds.server.http.BaseHttpServer;
-
+import java.util.HashMap;
+import java.util.Map;
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import org.apache.hadoop.hdds.conf.MutableConfigurationSource;
+import org.apache.hadoop.hdds.server.http.BaseHttpServer;
+import org.apache.hadoop.hdds.server.http.ServletElementsFactory;
+import org.apache.hadoop.security.SecurityUtil;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.apache.hadoop.security.authentication.server.AuthenticationFilter;
+import org.eclipse.jetty.servlet.FilterHolder;
+import org.eclipse.jetty.servlet.FilterMapping;
+import org.eclipse.jetty.servlet.ServletHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.apache.hadoop.ozone.s3.S3GatewayConfigKeys.OZONE_S3G_HTTP_BIND_HOST_KEY;
+import static org.apache.hadoop.ozone.s3secret.S3SecretConfigKeys.OZONE_S3G_SECRET_HTTP_AUTH_TYPE;
+import static org.apache.hadoop.ozone.s3secret.S3SecretConfigKeys.OZONE_S3G_SECRET_HTTP_ENABLED_KEY;
+import static org.apache.hadoop.ozone.s3secret.S3SecretConfigKeys.OZONE_S3G_SECRET_KEYTAB_FILE;
+import static org.apache.hadoop.ozone.s3secret.S3SecretConfigKeys.OZONE_S3G_SECRET_HTTP_KERBEROS_PRINCIPAL;
 
 /**
- * S3 Gateway specific configuration keys.
+ * Http server to provide S3-compatible API.
  */
 public class S3GatewayHttpServer extends BaseHttpServer {
 
+  private static final Logger LOG =
+      LoggerFactory.getLogger(S3GatewayHttpServer.class);
+
   /**
    * Default offset between two filters.
    */
   public static final int FILTER_PRIORITY_DO_AFTER = 50;
 
-  public S3GatewayHttpServer(MutableConfigurationSource conf,
-      String name) throws IOException {
+  public S3GatewayHttpServer(MutableConfigurationSource conf, String name)
+      throws IOException {
     super(conf, name);
     addServlet("icon", "/favicon.ico", IconServlet.class);
+    addSecretAuthentication(conf);
+  }
+
+  private void addSecretAuthentication(MutableConfigurationSource conf)
+      throws IOException {
+
+    if (conf.getBoolean(OZONE_S3G_SECRET_HTTP_ENABLED_KEY, false)) {
```

Comment:
Please define a constant variable for this default "false" value.

Reply:
I prefer use the constant variable. If you find any existing code which violates this code style, it's up to you to refactor it or not. But for the newly added code, I would suggest we follow the best practice coding style.